src/core/catalog.js

```diff
@@ -86,6 +86,14 @@ class Catalog {
     return shadow(this, "version", version.name);
   }
 
+  get needsRendering() {
```

Comment:
Can we please just document this getter instead, rather than adding a comment below, since that seems more helpful overall:
```javascript
  /**
   * @type {boolean} `true` for pure XFA documents,
   *   `false` for XFA Foreground documents.
   */
  get needsRendering() {
```